monitor/api/models/models.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os
+from typing import Dict, Optional
+
+from clients.api.api import APIClient
+from monitor.api.models.schema import ModelCoverageSchema, ModelSchema, NormalizedModelSchema
+from utils.json_utils import try_load_json
+
+YAML_FILE_EXTENSION = ".yml"
+SQL_FILE_EXTENSION = ".sql"
+
+
+class ModelsAPI(APIClient):
+    def get_models(self) -> Dict[str, NormalizedModelSchema]:
+        models_results = self.dbt_runner.run_operation(macro_name="get_models")
+        models = dict()
+        if models_results:
+            for model_result in json.loads(models_results[0]):
+                model_data = ModelSchema(**model_result)
+                normalized_model = ModelsAPI._normalize_dbt_model_dict(model_data)
+                model_unique_id = normalized_model.unique_id
+                models[model_unique_id] = normalized_model
+        return models
+    
+    def get_sources(self) -> Dict[str, NormalizedModelSchema]:
+        sources_results = self.dbt_runner.run_operation(macro_name="get_sources")
+        sources = dict()
+        if sources_results:
+            for source_result in json.loads(sources_results[0]):
+                source_data = ModelSchema(**source_result)
+                normalized_source = self._normalize_dbt_model_dict(source_data, is_source=True)
+                source_unique_id = normalized_source.unique_id
+                sources[source_unique_id] = normalized_source
+        return sources
+    
+    def get_coverages(self) -> Dict[str, ModelCoverageSchema]:
+        coverage_results = self.dbt_runner.run_operation(macro_name="get_coverages")
+        coverages = dict()
+        if coverage_results:
+            for coverage_result in json.loads(coverage_results[0]):
+                coverages[coverage_result["model_unique_id"]] = ModelCoverageSchema(
+                    table_tests=coverage_result["table_tests"],
+                    column_tests=coverage_result["column_tests"]
+                )
+        return coverages
+
+    @staticmethod
+    def _normalize_dbt_model_dict(model: ModelSchema, is_source: bool = False) -> NormalizedModelSchema:
+        model_name = model.name
+
+        owners = model.owners
+        if owners:
+            loaded_owners = try_load_json(owners)
+            if loaded_owners is not None:
+                owners = loaded_owners
+            else:
+                owners = [owners]
+
+        tags = model.tags
+        if tags:
+            loaded_tags = try_load_json(tags)
+            if loaded_tags is not None:
+                tags = loaded_tags
+            else:
+                tags = [tags]
+        
+        normalized_model = dict(model)
```

Comment:
I will parse it into json and then load it for creating a deep copy.
I am really against changing the original object to prevent weird behaviours if used in several places.
Plus we are creating a normalized model which is a different thing :) 
(In the future we can pass the normalization into the schema itself)